pinot-spi/src/main/java/org/apache/pinot/spi/stream/MessageBatch.java

```diff
@@ -96,6 +96,10 @@ default boolean isEndOfPartitionGroup() {
     return false;
   }
 
+  default boolean hasMissingOffsets() {
```

Comment:
Shall we call it `hasDataLoss()` which IMO is more intuitive. Also please add some javadoc